### DISCUSSION TITLE
HTTPS Probe support

### DISCUSSION THREAD

**Discussion by Slepwin** [loxilb-io/loxilb#217] [Ideas]
Hi,

It would be great if loxilb would be support for HTTPS probe/health-check type, because due security reason all network connections moving to encrypted approach and enpoints only answers to trusted LB (signed trusted CA).
API and CMD for CSR and private key generation would also good additions.

**Comment by UltraInstinct14** [comment]
@Slepwin Support for the requested feature has been [added](https://github.com/loxilb-io/loxilb/issues/220). We are undergoing some further tests and will soon have a minor release for this.  More documentation of https probing can be found [here](https://github.com/loxilb-io/loxilbdocs/blob/main/docs/cmd.md#endpoint).

**Comment by Slepwin** [comment]
Great job! Only one question: why rootCA cert is used and not dedicated certificate (web auth client) and how it works without private key file?

**Reply by nik-netlox** [reply]
Hi @Slepwin, 
For simple TLS, only CACert is required at the client side. Private key and private cert are required for mutual/strict TLS verification. Currently, we added support for only simple HTTPS probing. Do you have a specific requirement for probing with mutual TLS?

**Comment by nik-netlox** [comment]
@Slepwin , all the requested features have been  added in the latest release. Please feel free to check them out.
https://github.com/loxilb-io/loxilb/discussions/227
